### ISSUE TITLE
Update getModifierState browser support comments

### ISSUE THREAD

**Issue by nhunzaker** [facebook/react#13916]
We have a few stale comments about getEventModifierState:

https://github.com/facebook/react/blob/663835a43a5a92ff63c3e3b43e4c9ff40dde173b/packages/react-dom/src/events/getEventModifierState.js#L24-L26

IE8 isn't an issue anymore, [but Safari probably still is](https://caniuse.com/#feat=keyboardevent-getmodifierstate). It'd be nice to figure out specifically why we still need this module and update the comment with that info.

**Comment by nhunzaker** [comment]
Not at all :) Thanks!

**Comment by powens** [comment]
As you mentioned, it appears that the only issue is older versions of Safari (<= 10, according to he [caniuse link above](https://caniuse.com/#feat=keyboardevent-getmodifierstate)). Probably overkill, but I tested a modified create-react-app on Safari 7.1, and the modifierKeyToProp mapping object was used.

It also appears that Chrome now supports Lock-keys. I've removed that note as well.

Let me know if the changes are okay :)
